Header and Footer formatted

JessSumner opened the pull request:
Header has been simplified and formatted.
Navigation bar has a functioning navigation bar and is formatted.
Footer has been added and formatted.

JessSumner committed:
Additions to the header.

How:

Margins removed from the menu bar so it now spans the entire width.
Picture of Bubbles added to the header and centered. Gradient added over the image so it fades on the left and right.
Text removed as it was redundant with the logo.
Picture of Bubbles and logo inline with each other.

JessSumner committed:
Dropdown menu added using CSS. Not quite appearing in the right place yet.

JessSumner committed:
Navigation bar complete

How: Position centered and dropdown working using CSS.

JessSumner committed:
Footer complete

How:
Link to facebook page formatted and working. Copyright text added.

JessSumner committed:
Dropdown styled better but page width larger than screen and contents.

How:
Added blue background for the dropdown to match the navigation bar. Dropdown menu items have padding to space them out.

JessSumner committed:
Fixed width issue

How:
Navigation menu had been shifted across by 30% in order to center the links. The width was changed from 100% to 70% to compensate for this shift.

JessSumner commented:
@halogenandtoast If you have a moment and want to see what I've been up to I would greatly appreciate any feedback. Thank you

JessSumner committed:
Corrections suggested by Hound

Properties for header and footer css ordered alphabetically. HTML tags given classes rather than ids. White and transparent blue extracted into variables in custom.css.scss. Reduced the selectivity depth of the dropdown css.

JessSumner committed:
Hound corrections cont.

transparent extracted into a variable (used in the gradient for header image). Indentation fixed for one property. footer tag removed from the custom css file as it has it's own file.

halogenandtoast commented on app/assets/stylesheets/custom.css.scss:
```diff
@@ -2,40 +2,9 @@ $blue: #65BEDB;
-}
+$transparent-blue: transparentize($blue, 0.2);
+$white: #ffffff;
+$transparent: transparent;
```
not sure there's any value in assigning this to a variable here, I think it would be unlikely for you to change transparent to something else.

JessSumner replied:
I agree with you on this. The reason I assigned it to a variable was because it was picked up by Hound and I feel that a mistake I keep making is not assigning variables which has led to mystery guests. Therefore I felt it was important to assign more variables rather than less. I know there is a balance to find and I am quite on the fence with this one.

halogenandtoast commented on app/views/layouts/_header.html.erb:
```diff
@@ -1,18 +1,25 @@
-          <option><%= link_to "Volunteer", '#' %></option>
-          <option><%= link_to "Donate", '#' %></option>
-        </select>  
+      <li><%= link_to "How to help", "#" %></a>
```
The `</a>` **shouldn't** be required here unless I am missing something.

halogenandtoast commented on app/views/layouts/_header.html.erb:
```diff
@@ -1,18 +1,25 @@
-    <ul>
+  <div class="banner-container">
+    <div class="banner">
+      <%= image_tag("Logo Blue.jpg", class: "logo") %>
```
I didn't notice this before but in general it's good practice to keep filename lowercase and without spaces. I'd rename this file in `app/assets/images` to `logo_blue.jpg` and then call it here. You're not really doing anything wrong here, BUT filesystems treat uppercase letters differently (some ignore them and some don't) so it's generally best practice to stick with lower case for these types of things.

halogenandtoast commented on app/views/layouts/_header.html.erb:
```diff
@@ -1,18 +1,25 @@
+  <div class="banner-container">
+    <div class="banner">
+      <%= image_tag("Logo Blue.jpg", class: "logo") %>
+      <%= image_tag("Bubbles.jpg", class: "banner-image") %>
```
Same here.

MattMSumner commented on app/assets/stylesheets/header.css.scss:
```diff
@@ -0,0 +1,115 @@
+.gradient {
+  @include linear-gradient(to right, rgba(255, 255, 255, 1) 0%, 
+    rgba(255, 255, 255, 0) 5%, rgba(255, 255, 255, 0) 95%, 
+    rgba(255, 255, 255, 1) 100%, $fallback: $transparent);
```
this indentation threw me for a second. How would you feel about moving the `);` onto the next line? And giving each argument it's own line?

``` scss
  @include linear-gradient(
    to right,
    rgba(255, 255, 255, 1) 0%, 
    rgba(255, 255, 255, 0) 5%,
    rgba(255, 255, 255, 0) 95%, 
    rgba(255, 255, 255, 1) 100%,
    $fallback: $transparent
  );
```

MattMSumner commented:
Few comments otherwise this LGTM.

JessSumner committed:
Changes following feedback from PR

Images file names cahnged to be lowercase, no spaces and more descriptive. Changed the formating of the gradient css to be clearer.

JessSumner committed:
Changed color variables.

Color variables written lowercase.
